### PR TITLE
RFR: Added test cases from polarion to automate

### PR DESCRIPTION
**Pull request by ganeshhubale**
Signed-off-by: Ganesh Hubale <ghubale@redhat.com>

@sshveta 
- can you please check if you want this as a test case - https://issues.redhat.com/browse/WINDUP-2940. There is test for it in polarion.
- can you please provide test steps for this one -https://issues.redhat.com/browse/WINDUPRULE-697, Because I cant see `cloud-readiness` in a target.

### CODE REVIEW

**Review by nitishSr — APPROVED** [review]
LGTM

**Review by sshveta — CHANGES_REQUESTED** [review]

File: mta/tests/test_global_rules.py [hunk 1]
```diff
@@ -128,6 +128,7 @@ def test_invalid_rule_file_type(application, request):
         expectedResults:
             1. Invalid global_1 custom rule file should have 0 rules
     """
+    # TODO(ghubale): Upload an empty xml and error should be handled
```
[the remark this answers is not in the document]
Comment (ganeshhubale) [reply mta/tests/test_global_rules.py:131]: Done

File: mta/tests/test_global_rules.py [hunk 2]
```diff
@@ -196,3 +197,21 @@ def test_filter_global_system_rule(application):
...
+        initialEstimate: 1/12h
+        caseimportance: medium
+        testSteps:
+            1. Login to MTA web console
+            2. Navigate to Global > Rules configuration > Custom rules
+            3. Click on Add rule button and got to server path tab and browse rules folder
+            4. Click on Close button
+        expectedResults:
+            1. Error should be handled
+    """
```
[the remark this answers is not in the document]
Comment (ganeshhubale) [reply mta/tests/test_global_rules.py:220]: @sshveta MTA application is not recognizing servere path given from FTP servere.

[the remark this answers is not in the document]
Comment (ganeshhubale) [reply mta/tests/test_global_rules.py:220]: Added folder with rule files but unable to upload - ftp://rhev-node-09.rdu2.scalelab.redhat.com/zpool01/mnt/mta/applications/valid_invalid_rules/

[the remark this answers is not in the document]
Comment (ganeshhubale) [reply mta/tests/test_global_rules.py:222]: It worked with putting ```/tmp``` as server rules path.

File: mta/entities/global_config/rules_configuration.py [hunk 3]
```diff
@@ -155,17 +172,34 @@ def __init__(self, application, file_name):
         self.application = application
         self.file_name = file_name
 
-    def upload_custom_rule_file(self):
+    def upload_custom_rule_file(self, server_path=False, scan_recursive=False):
         """Method for uploading custom rule file
+
+        server_path: if True then upload rule file by server path
+        dir_path: Provide path to folder/file containing rule files
```
Comment (sshveta) [inline mta/entities/global_config/rules_configuration.py:179]: This is not used , please remove

File: mta/tests/test_global_rules.py [hunk 4]
```diff
@@ -111,42 +113,47 @@ def test_analysis_global_custom_rule(application, add_global_custom_rule, create
...
+        rules_configurations = CustomRulesConfiguration(application, file_name)
+        rules_configurations.upload_custom_rule_file()
+        view = rules_configurations.create_view(CustomRulesView)
+        view.table.wait_displayed("20s")
+
+        all_rules = view.table.read()
+        for rule in all_rules:
+            if rule["Short path"] == file_name:
+                assert int(rule["Number of rules"]) == 0
+    else:
```
Comment (sshveta) [inline mta/tests/test_global_rules.py:151]: what is this for ?
assert False ?

**Review by sshveta — CHANGES_REQUESTED** [review]
Looks good @ganeshhubale , minor comments.

File: mta/tests/test_global_rules.py (hunk 4 above)
Comment (ganeshhubale) [reply mta/tests/test_global_rules.py:151]: Case:
`all_rules` list empty. In this case, for loop fails and test has to FAIL.

**Review by sshveta — APPROVED** [review]

**Pull request merged**